src/app/note-list/note-list.component.html

```diff
@@ -0,0 +1,39 @@
+<div class="flex">
+  <button mat-raised-button color="primary" (click)="addNote.emit()">
+    Add Note
+  </button>
+</div>
+<br>
+<table mat-table [dataSource]="noteList" class="mat-elevation-z8">
+  <ng-container matColumnDef="id">
+    <th mat-header-cell *matHeaderCellDef> ID </th>
+    <td mat-cell *matCellDef="let note"> {{note.id}} </td>
+  </ng-container>
+
+  <ng-container matColumnDef="name">
+    <th mat-header-cell *matHeaderCellDef> Note Name </th>
+    <td mat-cell *matCellDef="let note"> {{note.name}} </td>
+  </ng-container>
+
+  <ng-container matColumnDef="note">
+    <th mat-header-cell *matHeaderCellDef> Note </th>
+    <td mat-cell *matCellDef="let note"> {{note.note.substring(0, 20) + "..."}} </td>
```

Comment:
Using functions is generally not a good thing in templates like this because they can be called many times per second. Generally for this kind of stuff you would use a pipe so that it's memoized